optuna/samplers/_tpe/sampler.py

```diff
@@ -554,7 +607,12 @@ def _get_observation_pairs(
     param_names: List[str],
     multivariate: bool,
     constant_liar: bool = False,  # TODO(hvy): Remove default value and fix unit tests.
-) -> Tuple[Dict[str, List[Optional[float]]], List[Tuple[float, List[float]]]]:
+    get_constraints: bool = False,
```

Comment:
On the other hand, do we need this argument? I think we can just return `List[Optional[Sequence[float]]]` always.

Reply:
I'm wondering if it is fine to return something like `[None, None, ...]`, which we do not use at all, when `constraints_func` is `None`.
But it might be probably better than the current implementation because the interface is consistent?

Reply:
Also, I give the return value of this function to `_split_observation_pairs` and check if it is `None`. If we always return `List[Optional[Sequence[float]]]`, we might need to add an additional argument to `_split_observation_pairs` in order to specify if we use constrains.
https://github.com/optuna/optuna/blob/1e2bca1bd742935d9463206a30704f5b6c4bec96/optuna/samplers/_tpe/sampler.py#L714

Reply:
So, I think it's okay to return `Optional[List[Optional[Sequence[float]]]]` by this function, although a little bit complicated. What do you think? @HideakiImamura

Reply:
Thanks for the explanation. It's OK to leave the type. Thanks.